plugins/cachekey/cachekey.cc

```diff
@@ -445,49 +500,35 @@ CacheKey::appendPath(Pattern &pathCapture, Pattern &pathCaptureUri)
 void
 CacheKey::appendHeaders(const ConfigHeaders &config)
 {
-  if (config.toBeRemoved() || config.toBeSkipped()) {
-    // Don't add any headers to the cache key.
-    return;
-  }
-
-  TSMLoc field;
-  StringSet hset; /* Sort and uniquify the header list in the cache key. */
-
-  /* Iterating header by header is not efficient according to comments inside traffic server API,
-   * Iterate over an 'include'-kind of list to avoid header by header iteration.
-   * @todo: revisit this when (if?) adding regex matching for headers. */
-  for (StringSet::iterator it = config.getInclude().begin(); it != config.getInclude().end(); ++it) {
-    String name_s = *it;
-
-    for (field = TSMimeHdrFieldFind(_buf, _hdrs, name_s.c_str(), name_s.size()); field != TS_NULL_MLOC;
-         field = ::nextDuplicate(_buf, _hdrs, field)) {
-      const char *value;
-      int vlen;
-      int count = TSMimeHdrFieldValuesCount(_buf, _hdrs, field);
-
-      for (int i = 0; i < count; ++i) {
-        value = TSMimeHdrFieldValueStringGet(_buf, _hdrs, field, i, &vlen);
-        if (value == nullptr || vlen == 0) {
-          CacheKeyDebug("missing value %d for header %s", i, name_s.c_str());
-          continue;
-        }
-
-        String value_s(value, vlen);
+  if (!config.toBeRemoved() && !config.toBeSkipped()) {
```

Comment:
Why was the original logic reversed and the early return removed here?

Reply:
Before adding `--capture-header` list there was only code that handled `--include-header` list. `toBeRemoved()` and `toBeSkipped()` are associated with `--capture-header` list processing so exiting the function earlier will not give `--capture-header` list related code to run. 

Beforehand considered various ways to add the new `--capture-header` functionality (parameters/design-wise) and it seems there might be a moment in time when I will redesign or refactor things to generalize various lists handling in the plugin but with this implementation I think we are still in a good shape (unless we find something problematic or not performing well).

Reply:
This makes sense to me. Thanks.